### PR TITLE
borrow more user action

### PR DESCRIPTION
**Pull request by graykode**

### CODE REVIEW

File: components/modal/borrow-more-modal.tsx [hunk 1]
```diff
@@ -34,59 +41,111 @@ const BorrowMoreModal = ({
   position,
   onClose,
 }: {
-  position: {
-    currency: Currency
-    amount: string
-  } | null
+  position: LoanPosition
```
Comment (onemonster) [inline components/modal/borrow-more-modal.tsx:44]: `LoanPosition | null`

File: components/modal/borrow-more-modal.tsx [hunk 2]
```diff
@@ -34,59 +41,111 @@ const BorrowMoreModal = ({
...
   onClose: () => void
 }) => {
+  const { prices } = useCurrencyContext()
   const [value, setValue] = useState('')
+
+  const amount = useMemo(
+    () => (position ? parseUnits(value, position.underlying.decimals) : 0n),
```
Comment (onemonster) [inline components/modal/borrow-more-modal.tsx:51]: Let's
```
if (!position) {
    return <></>
}
```
And get rid of all null checks for the code below

File: components/modal/borrow-more-modal.tsx [hunk 3]
```diff
@@ -95,14 +154,35 @@ const BorrowMoreModal = ({
...
+          amount > available ||
+          amount + maxInterest > maxLoanAmountExcludingCouponFee
+        }
+        className="font-bold text-base sm:text-xl bg-green-500 disabled:bg-gray-100 dark:disabled:bg-gray-800 h-12 sm:h-16 rounded-lg text-white disabled:text-gray-300 dark:disabled:text-gray-500"
+        onClick={async () => {
+          console.log('borrowing')
+        }}
       >
-        Confirm
+        {amount === 0n
```
Comment (onemonster) [inline components/modal/borrow-more-modal.tsx:179]: `Enter loan amount` looks unnecessary.  Let's remove this one.

Comment (graykode) [reply components/modal/borrow-more-modal.tsx:179]: image
Nope I think it was needed

File: components/modal/borrow-more-modal.tsx (hunk 2 above)
Comment (graykode) [reply components/modal/borrow-more-modal.tsx:51]: image

I believe setting the 'amount' to zero would make it cleaner. Additionally, it appears that the withdrawal model logic is the same. You can refer to it here: [Link](https://github.com/clober-dex/coupon.finance/blob/dev/components/modal/withdraw-modal.tsx#L27)

File: components/modal/borrow-more-modal.tsx (hunk 1 above)
Comment (graykode) [reply components/modal/borrow-more-modal.tsx:44]: image

I've already implemented a null check here: [Link](https://github.com/clober-dex/coupon.finance/blob/feat/borrow-more-view/components/borrow.tsx#L424). It seems like fixing all occurrences of position potentially being undefined is a critical safety measure.

File: components/modal/borrow-more-modal.tsx (hunk 3 above)
Comment (onemonster) [reply components/modal/borrow-more-modal.tsx:183]: I was thinking about using custom messages only for error cases, but this looks good too! Let's talk with Luna about what UX is better before launch.

There are three different states.
1. Enabled with Default Message (ex: Confirm)
2. Disabled due to suggestion (ex: Enter loan amount)
3. Disabled due to error state (ex: Insufficient ETH balance)

I was thinking about differentiating between 2 and 3. Let's debate over this in the future. Looks good for now.

File: components/modal/borrow-more-modal.tsx (hunk 1 above)
Comment (onemonster) [reply components/modal/borrow-more-modal.tsx:44]: I missed that. Looks good! Shouldn't we use `position?` -> `position` then?

File: components/modal/borrow-more-modal.tsx (hunk 2 above)
Comment (onemonster) [reply components/modal/borrow-more-modal.tsx:51]: Isn't position never null?

Comment (graykode) [reply components/modal/borrow-more-modal.tsx:53]: Yeah!

**Comment by graykode** [comment]
@onemonster Let's approve and merge it!

**Review by onemonster — APPROVED** [review]

**Pull request merged**